jobserver/management/commands/add_osi_report.py

```diff
@@ -0,0 +1,121 @@
+import hashlib
+import shutil
+import sys
+from datetime import datetime, timezone
+
+import requests
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from interactive.dates import END_DATE, START_DATE
+from interactive.models import AnalysisRequest
+from jobserver.models import Backend, User, Workspace
+
+
+BASE_URL = f"{settings.BASE_URL}/api/v2/releases/"
+
+
+def create_file(*, new_path):
+    report_path = settings.BASE_DIR / "outputs/report.html"
+    shutil.copy(report_path, new_path)
+
+
+def create_release(*, backend, path, user):
+    headers = {
+        "Authorization": backend.auth_token,
+        "Content-Type": "application/json",
+        "OS-User": user.username,
+    }
+    payload = {
+        "files": [
+            {
+                "name": path.name,
+                "url": "http://localhost:8001",
+                "size": path.stat().st_size,
+                "sha256": hashlib.sha256(path.read_bytes()).hexdigest(),
+                "date": datetime.fromtimestamp(
+                    path.stat().st_mtime, tz=timezone.utc
+                ).isoformat(),
+                "metadata": {},
+                "review": {
+                    "status": "APPROVED",
+                    "comments": {},
+                },
+            }
+        ],
+        "metadata": {},
+        "review": {},
+    }
+    url = f"{BASE_URL}workspace/opensafely-internal-interactive"
+    r = requests.post(url, json=payload, headers=headers)
+    r.raise_for_status()
+
+    return r.json()["release_id"]
+
+
+def submit_request(*, backend, user, workspace):
+    # create a JobRequest and AnalysisRequest to mirror a user making a request
+    job_request = workspace.job_requests.create(
+        backend=backend,
+        created_by=user,
+        sha="",
```

Comment:
Oh, I've just had an interesting thought.. sorry, wish I'd realised this earlier..

We need to know the Analysis Request ID in order to make the commit, because it will need to be added to the project.yaml.

This means we need to create and save the Analysis Request, before we make the Job Request.

So, AnalysisRequest.job_request needs to be nullable and the `AnalysisRequest.objects.create` should be done first.

Walking skeletons are useful, I guess?

Reply:
oh no 😱

I think we can avoid a null by wrapping this code in the view in a transaction, so we'll create an AnalysisRequest instance with `AnalysisRequest(…)`, create and save the JobRequest, update the AnalysisRequest instance and save it to the db.  I think we'll need to see how the implementation of `create_commit()` plays out before we make any changes with this or the view though!